### PR TITLE
Global topics

### PR DESCRIPTION
**Pull request by mrtcode**

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -45,7 +45,10 @@ module.exports = function (onInit) {
 	var server;
 	var statusIntervalID;
 	var stopping;
-	
+	var globalTopics = [
+		'styles',
+		'notifications'
```
Comment (dstillman) [inline server.js]: 'translators', not 'notifications'

File: server.js [hunk 2]
```diff
@@ -65,10 +68,19 @@ module.exports = function (onInit) {
 		
 		switch (data.event) {
 		case 'topicUpdated':
-			connections.sendEventForTopic(topic, event, {
-				topic: topic,
-				version: data.version
-			});
+			if (globalTopics.indexOf(topic) >= 0) {
```
Comment (dstillman) [inline server.js]: I think we should be able to use `includes()` now.

File: server.js [hunk 3]
```diff
@@ -65,10 +68,19 @@ module.exports = function (onInit) {
...
+				setTimeout(function () {
+					connections.sendEventForTopic(topic, event, {
+						topic: topic,
+						version: data.version
+					});
+				}, 30000);
```
Comment (dstillman) [inline server.js]: Let's move the delay to a constant at the top.

File: server.js [hunk 4]
```diff
@@ -65,10 +68,19 @@ module.exports = function (onInit) {
...
 		case 'topicUpdated':
-			connections.sendEventForTopic(topic, event, {
-				topic: topic,
-				version: data.version
-			});
+			if (globalTopics.indexOf(topic) >= 0) {
+				setTimeout(function () {
+					connections.sendEventForTopic(topic, event, {
+						topic: topic,
+						version: data.version
```
Comment (dstillman) [inline server.js]: There won't be a version for these.

File: server.js [hunk 5]
```diff
@@ -277,6 +291,17 @@ module.exports = function (onInit) {
 			if (topics && topics.length) {
 				for (let j = 0; j < topics.length; j++) {
 					let topic = topics[j];
+					// Allow global topics
+					if (globalTopics.indexOf(topic) >= 0) {
```
Comment (dstillman) [inline server.js]: `includes()`

**Comment by mrtcode** [comment]
It seems that subscriptionsDeleted action is optimized for apiKeys. Therefore some responses that don't have an apiKey now look a bit strange:
```
-> {"action":"deleteSubscriptions","subscriptions":[{"topic":"translators"},{"topic":"styles"}]}
<- {"event":"subscriptionsDeleted","subscriptions":[{"topics":["translators"]},{"topics":["styles"]}]}
```

Maybe we should join public and global topics and return:
```
<- {"event":"subscriptionsDeleted","subscriptions":[{"topics":["translators", "styles"]}]}
```

OR maybe we should just return an array of objects where each of them consists of one topic (and apiKey). It would make response more intuitive:
```
<- {"event":"subscriptionsDeleted","subscriptions":[{"topic":"translators"},{"topic":"styles"},{"topic":"styles", "apiKey": "qrheogiejrgijeroejrg"}]}
```

Because current deleteSubscriptions query doesn't group topics by key either:
```
-> {"action":"deleteSubscriptions","subscriptions":[{"apiKey":"asdf","topic":"/users/123456"},{"apiKey":"asdf","topic":"/users/234567"}]}
```

**Comment by dstillman** [comment]
```
<- {"event":"subscriptionsDeleted","subscriptions":[{"topic":"translators"},{"topic":"styles"},{"topic":"styles", "apiKey": "qrheogiejrgijeroejrg"}]}
```
Yeah, I think this makes sense. It's sort of awkward either way — you have to specify topics individually, but you can pass a single API key to delete multiple topics. But still probably cleanest to return the individual pairs that were deleted.

**Pull request merged**